### DISCUSSION TITLE
Jan2022 updated prebuilt firmware not working

### DISCUSSION THREAD

**Discussion by axg20202** [phil-barrett/grblHAL-teensy-4.x#41] [Q&A]
Hi Phil. I get a constant 'connecting to controller' message in IOSender after flashing the latest prebuilds. Reverting back to the previous version works fine. By the way, the README file for the prebuilds says there are no longer seperate USB and Ethernet versions, but there are two prebuild zips in the repo, one for USB and another for Ethernet. Have tried both of these and neither allows connection via IOSender.

**Comment by phil-barrett** [comment]
Hmmm, which hex file did you use?

The two prebuilt zips contain the same files.  That was in case someone had linked to them.

**Comment by phil-barrett** [comment]
Haven't heard back from you. I downloaded configurations.zip, extracted all the .hex files and tested them.  In each case, they worked and I was able to connect via ioSender 2.0.35 (current release).  Please download configurations.zip and try again.  There was one previous 
instance of file corruption on github so perhaps your download was somehow damaged?

**Reply by axg20202** [reply]
Ok, Phil, thanks for checking the files. I will download the files and try again.

**Comment by phil-barrett** [comment]
Also, make sure you don't have any other program connected to the teensy.

**Reply by axg20202** [reply]
OK, just retried. No change. Note that there is nothing connected to the board. Unlike the older firmware version, when I launch IOSender with the latest hex file installed I immediately get an E-stop warning that returns a few seconds later when I dismiss it and the repeated message is asking me to quit the programme all together. Meanwhile, the info bar at the bottom continues to say "Waiting for controller....". Perhaps a recent revision to GRBLHAL is preventing the board from starting up with nothing else connected?

**Comment by phil-barrett** [comment]
OK, what you need to do is short the Halt terminal to ground.  It is seeing EStop being asserted.  Details are [here](https://github.com/terjeio/grblHAL/wiki/First-Run-Grbl-Settings).

**Reply by axg20202** [reply]
Thanks Phil, that sorted it. Strange that the previous premade version didn't need this. Bits set to inverted already maybe.

**Reply by phil-barrett** [reply]
You may have changed them earlier and the new .hex wiped out the old settings. I should have caught that sooner.

**Reply by axg20202** [reply]
No changes were made. Old hex as downloaded swapped out with new; switching them back and forth resolved/recreated the error.

**Reply by phil-barrett** [reply]
Interesting.  Not sure when the defaults were changed.  Glad you got it all sorted out!  

Generally, in grblHAL, a new hex with different plug-ins or things like number of axes will revert settings to default. Something to keep in mind when upgrading.
